woocommerce/woocommerce
# [Experimental] Product Filters Redesign > Overlay: Add Fullscreen view

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #49977, #49933.

- This PR adds the functionality to open the Fullscreen View when shoppers click on the Product Filters Overlay Navigation button.
- When the dialog is opened, shoppers will be able to see the content of the Product Filters Overlay template part.
- When clicking on the Close button, the dialog should be closed.



### How to test the changes in this Pull Request:



> [!NOTE]  
> **Pre-requisite**
> - If you're using WC Beta Tester plugin, make sure the `experimental-blocks` feature is enabled (Tools > WCA Test Helper > Features).
> - Make sure to reset the Product Filters (Experimental) template part.

1. Navigate to Product Catalog template (Appearance > Editor > Templates > Product Catalog template - or any archive template).
2. Click on "Edit" to open the WordPress editor.
3. In the editor, you should see a block inserter icon or an option to "Add Block". Click on it.
4. Search for "Product Filters (Experimental)" template part in the block inserter. Add it to your content.
5. On the Inspector Controls, set the Overlay mode to 'Always'.
6. Ensure the Product Filters Overlay Navigation block is visible in the editor.
7. Save the changes and go to the Shop page (`/shop`).
8. Ensure you can see the Product Filters Overlay Navigation block (the icon followed by the text "Filters").
9. Click on it and check that a dialog opens containing the content from the Product Filters Overlay template part.
10. Click on the Close button and make sure the dialog closes.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment  Add the Fullscreen view to the Product Filters

</details>
